javascript/UserForm.js

```diff
@@ -330,11 +330,16 @@
 					newRule.removeClass("hidden");
 
 					// update the fields dropdown
-					newRule.children("select.fieldOption").empty();
+					var optionChildren = newRule.children("select.fieldOption");
+					optionChildren.empty();
 
-					$("#Fields_fields li.EditableFormField").each(function (i, domElement) {
-						var name = $(domElement).attr("id").split(' ');
-						newRule.children("select.fieldOption").append("<option value='"+ name[2] + "'>"+ $(domElement).find(".text").val() + "</option>");
+					$("#Fields_fields li.EditableFormField").each(function () {
+						var name = $(this).attr("id").split(' ');
```

Comment:
[Here's a good place to start.](http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule)